spec/views/curriculum/key_stages/show_spec.rb

```diff
@@ -23,6 +23,14 @@
       expect(rendered).to have_css(".hero__heading", text: "Key stage 1 resources")
     end
 
+    it "links to the journey poster" do
+      key_stage = JSON.parse(unit_json, object_class: OpenStruct).data.key_stage.level
+      key_stage_type = ["1", "2"].include?(key_stage) ? "Primary" : "Secondary"
+      key_stage_range = (key_stage_type == "Primary") ? "1-2" : "3-4"
+
+      expect(rendered).to have_link("viewing and progressing through our #{key_stage_type} curriculum journey", href: "https://static.teachcomputing.org/journey/Curriculum.Journey_#{key_stage_type}.#{key_stage_range}.pdf")
+    end
```

Comment:
I'm generally not a fan of very complicated logic in tests like this, because we've essentially got a mirror of what the real code should be doing.

IMO tests shouldn't have any conditionals in them. They should construct the exact data required and then assert that the subject method/class/page does what you expect it to when given the particular data you've constructed and passed to it.

Could this be simplified so the link in that "have_link" assertion has the correct values just being part of the string?